notebooks/Notebook1.md

```diff
@@ -2,33 +2,94 @@
+
+In tandem with that I need some clear examples of the DSL. The feedback showed me that how the dsl will work also needs some defining. Is it something that rolls singular events? Or something that creates graphs? That is a question I need to answer. Ideally both but again we need to keep it realistic.
 
 ## Description
 
-**TODO:** Fill in this part with information about your work this week:
-important design decisions, changes to previous decisions, open questions,
-exciting milestones, preliminary results, etc. Feel free to include images
-(e.g., a sketch of the design or a screenshot of a running program), links to
-code, and any other resources that you think will help clearly convey your
-design process.
+So the first step of this week was working to create a step by step guide of 1 what I was making and 2 how I was going to do it. In advance I apologize for the format of these and specifically how the typing ends up looking. I have dyslexia and it can make some of my sentences really impossible to understand (I swap word order alot). Feel free to comment on those in reviews so I can fix them. 
+
+### Research: 
+Firstly Here are some documents that  I might return to in my implemntation of the project
+ - [Create an Exectuable Command Prompt Script](https://stackoverflow.com/questions/13975733/how-to-create-an-executable-command-prompt-script)
+ - [Executing Shell Commands in Python](https://stackabuse.com/executing-shell-commands-with-python/)
+ - [Convert Python To Executable](https://towardsdatascience.com/how-to-easily-convert-a-python-script-to-an-executable-file-exe-4966e253c7e9)
+ - [Scalapy -- Running Python packages in Scala](https://github.com/scalapy/scalapy)
+ - [Turning and Exporting a Scala Project as One single .jar file](https://github.com/sbt/sbt-onejar)
+ - [GUI programming in Scala](https://otfried.org/scala/gui.html)
+
+
+### Thoughts: 
+
+From my research the actual structure of my program, I chose to do the following: The DSL will be a External Domain language, and implemented in scala. While I did a internal one for the previous project, I will have some guidence from others that did external while I set that up. Then, the external DSL will be set into a simple GUI following the tutorial above. Im not going to focus on prettyness or anything like that, I imagine this will be simple and I will focus mostly on the implementation of the language. The gui will either allow directly editing a text document and a run button, or to select a already written text file. 
+
+The language will have two overarcing setups. 1 build a roll roll {description} and then enacting stats on that roll. Rolls can be reused by a name and can be nested if possible. See design for more. The creation of stats would use the Scalapy package to grab some python graphing packages. There might be some scala graphing packages tho, so Ill keep an eye out for that. 
+
```

Comment:
It looks like you can use plotly directly from Scala.

[https://plotly.com/scala/getting-started/](https://plotly.com/scala/getting-started/)